src/main/java/io/github/fusionflux/portalcubed/content/prop/Prop.java

```diff
@@ -0,0 +1,256 @@
+			lerpSteps = 0;
+			syncPacketPositionCodec(getX(), getY(), getZ());
+
+			var vel = getDeltaMovement();
+			if (getHeldBy().isEmpty()) {
+				if (!onGround() && !isNoGravity())
+					vel = vel.subtract(0, LivingEntity.DEFAULT_BASE_GRAVITY / (isInWater() ? 16 : 1), 0);
+				var posBelow = getBlockPosBelowThatAffectsMyMovement();
+				float f = level().getBlockState(posBelow).getBlock().getFriction();
+				f = onGround() ? f * .91f : .91f;
+				vel = new Vec3(vel.x * f, Math.max(vel.y, -TERMINAL_VELOCITY), vel.z * f);
+				setDeltaMovement(vel);
+				move(MoverType.SELF, vel);
+			} else {
+				var player = ((Player) level().getEntity(getHeldBy().getAsInt()));
+				var holdPoint = player.getEyePosition().add(Vec3.directionFromRotation(player.getXRot(), player.getYRot()).scale(2));
+				float holdYOffset = -getBbHeight() / 2;
+				move(MoverType.PLAYER, position().vectorTo(holdPoint.add(0, holdYOffset, 0)));
+				setYRot(-player.getYRot() % 360);
+				if (position().distanceToSqr(player.getEyePosition()) >= 4.5 * 4.5) {
+					drop(player);
+					((PlayerExt) player).pc$heldProp(OptionalInt.empty());
+				}
+			}
+		} else if (lerpSteps > 0) {
+			double delta = 1.0 / lerpSteps;
+			setPos(
+				Mth.lerp(delta, getX(), lerpX),
+				Mth.lerp(delta, getY(), lerpY),
+				Mth.lerp(delta, getZ(), lerpZ)
+			);
+			setYRot((float) Mth.rotLerp(delta, getYRot(), lerpYRot));
+
+			--lerpSteps;
+		}
+	}
+
+	@Override
+	public void lerpTo(double x, double y, double z, float yaw, float pitch, int interpolationSteps) {
+		lerpX = x;
+		lerpY = y;
+		lerpZ = z;
+		lerpYRot = yaw;
+		lerpSteps = interpolationSteps;
+	}
+
+	@Override
+	public InteractionResult interact(Player player, InteractionHand hand) {
+		var itemInHand = player.getItemInHand(hand);
+		var level = level();
+		if (type.hasDirtyVariant) {
+			if (player.getAbilities().mayBuild && itemInHand.is(PortalCubedTags.Item.AGED_CRAFTING_MATERIALS) && !getVariantFlag(dirtyFlagIndex())) {
+				if (!level.isClientSide) {
+					setVariantFlag(dirtyFlagIndex(), true);
+					level.playSound(null, this, SoundType.VINE.getPlaceSound(), SoundSource.PLAYERS, 1, .5f);
+					var particleOption = new BlockParticleOption(ParticleTypes.BLOCK, Blocks.VINE.defaultBlockState());
+					for (var dir : Direction.values()) {
+						double x = getX() + (dir.getStepX() * getBbWidth() / 2);
+						double y = getY() + (dir.getStepY() * getBbWidth() / 2);
+						double z = getZ() + (dir.getStepZ() * getBbWidth() / 2);
+						((ServerLevel) level).sendParticles(particleOption, x, y, z, level.random.nextInt(5, 8), 0, 0, 0, 1);
+					}
+					itemInHand.shrink(1);
+				}
+				return InteractionResult.sidedSuccess(level.isClientSide);
+			}
+		}
+		if (!isVehicle() && type == PropType.CHAIR) {
+			if (!level.isClientSide) player.startRiding(this);
+			return InteractionResult.sidedSuccess(level.isClientSide);
+		}
+		return super.interact(player, hand);
+	}
+
+	@Override
+	public boolean hurt(DamageSource source, float amount) {
```

Comment:
always returns true even when nothing happened. Invulnerability should be handled by isInvulnerableTo.

Reply:
why was this hidden

Reply:
github moment